app/components/sidebars/main/main_sidebar.js

```diff
@@ -168,6 +168,9 @@ export default class ChannelSidebar extends Component {
     };
 
     selectChannel = (channel, currentChannelId) => {
+        if (!channel) {
```

Comment:
this will make the drawer stay open.

Also look at line 185 it says `this.closeChannelDrawer();` and it should be `this.closeChannelSidebar();`

Reply:
Are you sure about it `closeChannelSidebar`? I don't see a reference to it.

Yes, I intentionally added the check before. we are not switching the channel so, shouldn't it stay open?. I am 0/5 on this though.

Reply:
yes right, I got confused with `closeSettingSidebar` my bad.

I know we are not closing the channel but looking at this from the users point of view, I did tap on something and nothing is going on, no feedback whatsoever, what should be the right way to handle this that's another thing...

Options:
1. determine why is actually null and prevent that from happening.
2. Give some feedback to the user that the channel you are trying to switch to can't be done
3. close the sidebar (this is the least recommended option IMO)

Reply:
I really wish i knew why we have null channel objects. Anyway, for now, I went with 3.